terraform/implementation/ecs/main.tf

```diff
@@ -0,0 +1,207 @@
+  certificate_body = var.qc_tls_cert      # Public cert from Cloudflare
+}
+
+data "aws_caller_identity" "current" {}
+
+module "vpc" {
+  source  = "terraform-aws-modules/vpc/aws"
+  version = "5.16.0"
+
+  name            = local.vpc_name
+  cidr            = var.vpc_cidr
+  azs             = var.availability_zones
+  private_subnets = var.private_subnets
+  public_subnets  = var.public_subnets
+  # if internal is true, then the VPC will not have a NAT or internet gateway
+  enable_nat_gateway = var.internal ? false : true
+  single_nat_gateway = var.internal ? false : true
+  create_igw         = var.internal ? false : true
+  tags               = local.tags
+}
+
+module "ecs" {
+  source  = "CDCgov/dibbs-ecr-viewer/aws"
+  version = "0.3.0"
+
+  public_subnet_ids  = flatten(module.vpc.public_subnets)
+  private_subnet_ids = flatten(module.vpc.private_subnets)
+  vpc_id             = module.vpc.vpc_id
+  region             = var.region
+
+  owner        = var.owner
+  project      = var.project
+  tags         = local.tags
+
+
+  phdi_version = "main"
+
+  service_data = {
+    query-connector = {
+      short_name     = "qc",
+      fargate_cpu    = 512,
+      fargate_memory = 1024,
+      min_capacity   = 1,
+      max_capacity   = 5,
+      app_repo       = "ghcr.io/cdcgov/dibbs-query-connector",
+      app_image      = "${terraform.workspace}-query-connector",
+      app_version    = "main",
+      container_port = 3000,
+      host_port      = 3000,
+      public         = true,
+      registry_url   = "${data.aws_caller_identity.current.account_id}.dkr.ecr.${var.region}.amazonaws.com",
+      env_vars = [
+        {
+          name  = "AWS_REGION",
+          value = var.region
+        },
+        {
+          name  = "HOSTNAME",
+          value = "0.0.0.0"
+        },        
+        {
+        name  = "fhir_url"
+        value = var.fhir_url
+        },
+        {    
+          name  = "cred_manager"
+          value = var.cred_manager
+        },
+        {    
+         name  = "DATABASE_URL"
+         value = "postgresql://${aws_db_instance.qc_db.username}:${aws_db_instance.qc_db.password}@${aws_db_instance.qc_db.endpoint}/${aws_db_instance.qc_db.db_name}"
+         },
+         {    
+         name  = "FLYWAY_URL"
+         value = "jdbc:postgresql://${aws_db_instance.qc_db.endpoint}/${aws_db_instance.qc_db.db_name}"
+
+         },
+         {    
+         name  = "FLYWAY_PASSWORD"
+         value = aws_db_instance.qc_db.password
+         },
+         {    
+         name  = "FLYWAY_USER"
+         value = aws_db_instance.qc_db.username
+         },
+         {    
+         name  = "UMLS_API_KEY"
+         value = var.umls_api_key
+         },
+                  {    
+         name  = "ERSD_API_KEY"
+         value = var.ersd_api_key
+         },
+      ]
+    }
+  }
+
+
+  # If intent is to pull from the phdi GHCR, set disable_ecr to true (default is false)
```

Comment:
Since this is used internally by the qc team and not a template used by others, the comments in this section could be removed.